askama_shared/src/parser.rs

```diff
@@ -864,6 +872,16 @@ fn block_let(i: &[u8]) -> IResult<&[u8], Node<'_>> {
 }
 
 fn block_for<'a>(i: &'a [u8], s: &'a Syntax<'a>) -> IResult<&'a [u8], Node<'a>> {
+    let else_block = preceded(
+        ws(tag("else")),
+        cut(tuple((
+            opt(tag("-")),
+            |i| tag_block_end(i, s),
+            |i| parse_template(i, s),
+            |i| tag_block_start(i, s),
+            opt(tag("-")),
+        ))),
+    );
```

Comment:
I think I'd prefer to have this in a separate function.

Reply:
Done.